be2-scala/src/main/scala/ch/epfl/pop/model/network/method/message/data/DataSchemaValidator.scala

```diff
@@ -0,0 +1,63 @@
+package ch.epfl.pop.model.network.method.message.data
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.JsonNode
+import ch.epfl.pop.pubsub.graph.Validator
+import com.networknt.schema.JsonSchema
+import scala.collection.JavaConverters._
+import ch.epfl.pop.model.network.method.message.data.ActionType.ActionType
+import ch.epfl.pop.model.network.method.message.data.ObjectType.ObjectType
+import org.slf4j.LoggerFactory
+import scala.util.{Try,Success,Failure}
+
+
+/**
+  * DataSchemaValidator Object, provides a validateSchema method that
+  * verifies a certain payload of a certain actionType and objectType
+  * is conform to the protocol.
+  */
+object DataSchemaValidator {
+  private final val objectMapper: ObjectMapper = new ObjectMapper()
+  private final val logger = LoggerFactory.getLogger("DataSchemaValidator")
+
+  /* Paths to Schemas */
+  private final val baseDir = "protocol/query/method/message/data/"
+  private final val dataCreateLaoPath = baseDir + "dataCreateLao.json"
+  private final val dataStateLaoPath = baseDir + "dataStateLao.json"
+  private final val dataUpdateLao = baseDir + "dataUpdateLao.json"
+
+  /* Validation Schemas */
+  //TODO: Add schemas for other features: Meetings, RollCalls...
+   private final lazy val createLaoSchema: JsonSchema = Validator.setupSchemaValidation(dataCreateLaoPath, objectMapper)
+   private final lazy val stateLaoSchema: JsonSchema  = Validator.setupSchemaValidation(dataStateLaoPath, objectMapper)
+   private final lazy val updateLaoSchema: JsonSchema = Validator.setupSchemaValidation(dataUpdateLao, objectMapper)
+
```

Comment:
You have some strange alignments in the file (here 1 space too many). Same line 38